doc/dune

```diff
@@ -1,8 +1,5 @@
 (rule
  (targets unreleased.rst)
- ; We need to keep this as the sphinx build still requires this file
- ; in-tree, to be fixed hopefully soon
```

Comment:
So what changed?

Reply:
I'm not sure, but this comment is quite old, so probably the build consolidation PR. I also do not know what it meant exactly (cc @ejgallego).

Reply:
Likely this was indeed from a time we had options to run sphinx both from make and from dune, should not be needed now.